tests/sentry/tasks/test_recap_servers.py

```diff
@@ -0,0 +1,197 @@
+        self.project_dos.update_option(RECAP_SERVER_OPTION_KEY, "http://example-dos.com")
+        self.project_tres.update_option(RECAP_SERVER_OPTION_KEY, "http://example-tres.com")
+
+        poll_recap_servers()
+
+        assert poll_project_recap_server.call_count == 3
+        poll_project_recap_server.assert_has_calls(
+            [call(self.project.id), call(self.project_dos.id), call(self.project_tres.id)],
+            any_order=True,
+        )
+
+    @pytest.mark.django_db
+    def test_poll_project_recap_server_incorrect_project(self):
+        poll_project_recap_server(1337)  # should not error
+
+    @pytest.mark.django_db
+    def test_poll_project_recap_server_missing_recap_url(self):
+        poll_project_recap_server(self.project.id)  # should not error
+
+    @pytest.mark.django_db
+    @patch("sentry.tasks.recap_servers.store_crash")
+    @responses.activate
+    def test_poll_project_recap_server_initial_request(self, store_crash):
+        payload = {
+            "results": 3,
+            "_embedded": {
+                "crash": [
+                    {"id": 1},
+                    {"id": 1337},
+                    {"id": 42},
+                ]
+            },
+        }
+        outgoing_recap_request = responses.add(
+            method=responses.GET,
+            url="http://example.com/rest/v1/crashes",
+            body=json.dumps(payload),
+            content_type="application/json",
+        )
+
+        self.project.update_option(RECAP_SERVER_OPTION_KEY, "http://example.com")
+        assert self.project.get_option(RECAP_SERVER_MOST_RECENT_POLLED_ID_KEY) is None
+
+        poll_project_recap_server(self.project.id)
+
+        assert outgoing_recap_request.call_count == 1
+        assert store_crash.call_count == 3
+        assert self.project.get_option(RECAP_SERVER_MOST_RECENT_POLLED_ID_KEY) == 1337
+
+    @pytest.mark.django_db
+    @patch("sentry.tasks.recap_servers.store_crash")
+    @responses.activate
+    def test_poll_project_recap_server_following_request(self, store_crash):
+        payload = {
+            "results": 3,
+            "_embedded": {
+                "crash": [
+                    {"id": 1},
+                    {"id": 1337},
+                    {"id": 42},
+                ]
+            },
+        }
+        # Encoded query: {8 TO *}
+        outgoing_recap_request = responses.add(
+            method=responses.GET,
+            url="http://example.com/rest/v1/crashes;q=id:%7B8%20TO%20%2A%7D",
+            body=json.dumps(payload),
+            content_type="application/json",
+        )
+        self.project.update_option(RECAP_SERVER_OPTION_KEY, "http://example.com")
+        self.project.update_option(RECAP_SERVER_MOST_RECENT_POLLED_ID_KEY, 8)
+
+        poll_project_recap_server(self.project.id)
+
+        assert outgoing_recap_request.call_count == 1
+        assert store_crash.call_count == 3
```

Comment:
what is this test asserting? Basically that you are hitting that very specific query?
Should some kind of filtering also happen? As in: We query for `id > 8`, but we ingest the crash with `id = 1` as well, even though by this logic that should already have happened.

Probably not a real world edge-case as the server should not give you such an id to begin with, but you have written your tests this way :-)

Reply:
Good catch. And you are correct, this test is only to assert that a valid url, with the encoded query, is hit. Will update mocked payload to not create any confusion.